src/state/cached_state.rs

```diff
@@ -81,6 +81,17 @@ impl<T: StateReader> CachedState<T> {
             .as_ref()
             .ok_or(StateError::MissingCasmClassCache)
     }
+
+    pub fn create_copy(&self) -> Self {
```

Comment:
we should avoid copying the state, it can be huge.

Reply:
And in any case, what would exactly be the difference with just calling `clone`?